pkg/agent/controller/serviceimport.go

```diff
@@ -191,13 +191,6 @@ func (c *ServiceImportController) serviceImportDeleted(key string) error {
 		c.endpointControllers.Delete(key)
 	}
 
-	labelSelector := labels.Set(map[string]string{lhconstants.LabelSourceName: si.Name}).AsSelector()
```

Comment:
Is the endpoint getting deleted due to the delete in the EndpointController or it is the garbage collector?

Reply:
I assume you mean the EndpointSlice. The EndpointController deletes it due to Endpoints deleted - which I assume would only happen if the service was deleted.  This isn't necessary b/c we set the ServiceImport as the owner which k8s will GC.

Reply:
ok got it.

Reply:
> Is the endpoint getting deleted due to the delete in the EndpointController or it is the garbage collector?

It gets deleted due to the owner ref that sets it to serviceimport.